modules/shared/stm32f2xx/inc/system_part3_loader.c

```diff
@@ -56,6 +57,10 @@ void module_system_part3_init()
 
 }
 
+void module_system_part3_newlib_impure_set(struct _reent* r, size_t size, uint32_t version, void* ctx) {
+    _impure_ptr = r;
```

Comment:
I'm wondering if we should add a run-time assert that would check that the size of the reent structure matches the size of the structure this module was built with – here and in the user part's callback, or, alternatively, have a static assert somewhere that would indicate that the size of the structure has changed between GCC releases.

Reply:
As per discussion, we'll leave this as is, as it doesn't guarantee anything.